lib/storage/bucket.js

```diff
@@ -257,6 +272,11 @@ Bucket.prototype.getFiles = function(query, callback) {
     callback = query;
     query = {};
   }
+  if (query.folder) {
+    query.delimiter = '/';
+    query.prefix = query.folder + '/';
```

Comment:
Is the prefix handler smart enough to handle `hello/world//` (i.e. multiple trailing slashes)? What about a leading slash?